index.js

```diff
@@ -1,35 +1,65 @@
-/*!
- * forwarded
- * Copyright(c) 2014 Douglas Christopher Wilson
- * MIT Licensed
- */
-
-/**
- * Module exports.
- */
+'use strict'
 
-module.exports = forwarded
+var Processor = require('./lib/processor')
+var schemas = require('./lib/schemas')
 
 /**
  * Get all addresses in the request, using the `X-Forwarded-For` header.
  *
- * @param {Object} req
+ * @param {http.IncomingMessage} req
  * @api public
  */
 
-function forwarded(req) {
+module.exports = function forwarded (req, options) {
+  options = options || {}
+
+  var opts = {
+    // default to only common + standard
+    // array order matters here
+    schemas: options.schemas || [
```

Comment:
imo it should only default to a single schema, not multiple. The reason is people are mainly going to be set up for one.